Splash/Splash/ShowersViewController.m

```diff
@@ -66,18 +88,37 @@ - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger
 
 - (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
 {
-    return self.showersArray.count;
+    return self.filteredData.count;
 }
 
 
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
     ShowerCell *cell = [self.showersTableView dequeueReusableCellWithIdentifier:@"showerCell"];
-    [cell setCell:self.showersArray[indexPath.section]];
+    [cell setCell:self.filteredData[indexPath.section]];
     cell.layer.cornerRadius = 25;
     cell.layer.borderWidth = 1;
     cell.layer.borderColor = [[UIColor grayColor] CGColor];
     return cell;
 }
+
+// called when text changes (including clear)
+- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
+    if (searchText.length != 0) {
+       
+            NSPredicate *predicate = [NSPredicate predicateWithBlock:^BOOL(Shower *evaluatedShower, NSDictionary *bindings) {
+                NSDateFormatter *format = [[NSDateFormatter alloc] init];
+                [format setLocalizedDateFormatFromTemplate:@"EEEE, MMM d, yyyy"];
+                return [[format stringFromDate:evaluatedShower.createdAt] containsString:searchText] || [[TimeViewController formatTimeString:[[NSString stringWithFormat:@"%@", evaluatedShower[@"len"]] intValue]] containsString:searchText];
```

Comment:
Reference method's of other viewcontroller's directly is not ideal. Let's move it to a helper class and reference it.